pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -123,10 +128,26 @@ type cloudWatchExecutor struct {
 	cfg         *setting.Cfg
 	sessions    SessionCache
 	regionCache sync.Map
+	logger      log.Logger
 
 	resourceHandler backend.CallResourceHandler
 }
 
+// instrumentContext adds plugin key-values to the context; later, logger.FromContext(ctx) will provide a logger
+// that adds these values to its output.
+// TODO: this really shouldn't be here!
```

Comment:
if it shouldn't be here, where should it be? Maybe if we know we can make a github issue for it and replace the comment with the reference to the github issue?

Reply:
That is a good question. For context, I think it doesn't belong here because it's solving what should be a universal problem for decoupled plugins. How are those other plugins solving or going to solve the problem? (I should look at that.) How important is it to do that in a unified way?

That said, I think it likely belongs in the sdk. But it might be nice to put it there in such a way that the [current middleware](https://github.com/grafana/grafana/blob/main/pkg/services/pluginsintegration/clientmiddleware/contextual_logger_middleware.go) can use it too.

I think that's enough to make an issue for reference, even if the solution isn't quite clear yet. I'll do that.

Reply:
(did that)